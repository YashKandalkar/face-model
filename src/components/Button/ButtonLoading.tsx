interface Props {
  className?: string;
}

const ButtonLoading = ({ className }: Props) => {
  return (
    <div className={className}>
      <span>
        <i />
        <i />
        <i />
      </span>
      <style jsx>{`
        div {
          align-items: center;
          background: #0256a6;
          border-radius: 0.25rem;
          cursor: default;
          display: flex;
          justify-content: center;
          min-height: 50px;
          outline: none;
          width: 100%;
        }
        span {
          align-items: center;
          display: flex;
          gap: 4px;
          justify-content: center;
          user-select: none;
        }
        i {
          animation: blink 1.4s infinite both;
          background-color: #fff;
          border-radius: 50%;
          display: inline-block;
          height: 5px;
          width: 5px;
        }
        i:nth-child(2) {
          animation-delay: 0.2s;
        }
        i:nth-child(3) {
          animation-delay: 0.4s;
        }
        @keyframes blink {
          0% {
            opacity: 0.2;
          }
          20% {
            opacity: 1;
          }
          100% {
            opacity: 0.2;
          }
        }
      `}</style>
    </div>
  );
};

export default ButtonLoading;
