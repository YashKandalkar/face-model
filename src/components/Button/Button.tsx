import React from "react";
import ButtonLoading from "./ButtonLoading";

interface Props {
  children?: React.ReactNode;
  className?: string;
  onClick?: React.MouseEventHandler<HTMLButtonElement>;
  loading?: boolean;
}

const Button = ({ children, className, onClick, loading }: Props) => {
  return loading ? (
    <ButtonLoading className={className} />
  ) : (
    <button onClick={onClick} className={className}>
      {children}
      <style jsx>{`
        button {
          background: #037bee;
          border-radius: 0.25rem;
          color: #f5f5f5;
          cursor: pointer;
          font-family: inherit;
          font-size: inherit;
          font-weight: inherit;
          min-height: 50px;
          line-height: 50px;
          outline: none;
          padding: 0 16px;
          transition: all 250ms cubic-bezier(0.4, 0, 0.2, 1);
          user-select: none;
          width: 100%;
        }
        button:hover,
        button:focus {
          background: #0256a6;
          outline: none;
        }
      `}</style>
    </button>
  );
};

export default Button;
