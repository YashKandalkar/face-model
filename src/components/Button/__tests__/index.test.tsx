import React from "react";
import { mount } from "enzyme";
import Button from "../index";

describe("Button", () => {
  it("should be defined", () => {
    expect(Button).toBeDefined();
  });

  it("should be rendered", () => {
    const wrapper = mount(<Button />);
    expect(() => wrapper.unmount()).not.toThrow();
  });

  it("should support loading", () => {
    const wrapper = mount(<Button loading={true} />);
    expect(() => wrapper.unmount()).not.toThrow();
  });

  it("should render children", () => {
    const wrapper = mount(<Button>Button</Button>);
    expect(wrapper.text()).toContain("Button");
  });

  it("should render different text", () => {
    const wrapper = mount(<Button>Text</Button>);
    expect(wrapper.text()).toContain("Text");

    wrapper.setProps({
      children: <span>文本</span>,
    });
    expect(wrapper.text()).toContain("文本");
  });

  it("should trigger callback function", () => {
    const WrapperButton = () => {
      const [state, setState] = React.useState<string>("state1");
      return <Button onClick={() => setState("state2")}>{state}</Button>;
    };
    const wrapper = mount(<WrapperButton />);
    expect(wrapper.text()).toContain("state1");

    wrapper.simulate("click");
    expect(wrapper.text()).toContain("state2");
  });

  it("should remove expired events", () => {
    const wrapper = mount(<Button>Event</Button>);
    wrapper.simulate("click");
    expect(() => wrapper.unmount()).not.toThrow();
  });

  it("should match snapshot", () => {
    expect(<Button>Snapshot</Button>).toMatchSnapshot();
    expect(<Button loading={true} />).toMatchSnapshot();
  });
});
