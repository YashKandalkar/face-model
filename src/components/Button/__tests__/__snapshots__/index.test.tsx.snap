// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match snapshot 1`] = `
<Button>
  Snapshot
</Button>
`;

exports[`Button should match snapshot 2`] = `
<Button
  loading={true}
/>
`;
